db/migrate/20190422185437_add_location_table.rb

```diff
@@ -0,0 +1,27 @@
+class AddLocationTable < ActiveRecord::Migration[5.1]
+  def up
+    create_table :locations, force: true do |t|
+      t.string :name, default: "", null: false, comment: "Full display name, such as a complete address"
+      t.string :level_name, default: "", null: false, limit: 20, comment: "Level of specificity (country, state, subdivision, or city)"
+      t.string :country_name, default: "", null: false, limit: 100, comment: "Country (or equivalent) of this location if available"
```

Comment:
Do we need all the `_name` suffixes?

Reply:
I think it's wise if we want to store `country_code` later... which begs the question of exactly which standardized format do we store in `country_name`?

Reply:
This is a bit of a cop-out answer, but I think all the tools we use are going to be based on OpenStreetMap data potentially mixed with other sources, so the fields will align with what's in the name:en field and https://wiki.openstreetmap.org/wiki/Nominatim/Country_Codes Maybe we can store ISO 3166-1 country_code to help disambiguate?

If we need to, it could be viable to have a curated list of countries we store ourselves, but it would probably get unwieldy for below country level.

Reply:
Storing ISO country codes in addition to full names seems wise.

Reply:
added :)